spec.bs

```diff
@@ -1226,6 +1222,282 @@ On the other hand, methods for getting data from the [=shared storage database=]
+    * `append`
+        - Required [=structured header/parameters=]:
+            * `key` ([=structured header/Token=], [=structured header/String=], or [=structured header/Byte Sequence=])
+            * `value` ([=structured header/Token=], [=structured header/String=], or [=structured header/Byte Sequence=])
+        - If the [=structured header/parameter=] value of `key` already exists in the [=shared storage database=] for the responding server's [=/origin=], then `append` updates the [=entry/key=]'s [=shared storage database/entry=]'s [=entry/value struct=]'s [=value struct/value=] by appending the [=structured header/parameter=] value for `value` to it.
+        - If the [=structured header/parameter=] value of `key` does not already exist in the [=shared storage database=] for the responding server's [=/origin=], `append` is equivalent to `set` with `ignore_if_present` false, i.e. `append` writes the [=shared storage database/entry=] consisting of the [=structured header/parameter=] values for `key` and `value`, as the [=shared storage database/entry=]'s [=entry/key=] and [=shared storage database/entry=]'s [=entry/value struct=]'s [=value struct/value=] respectively, in the [=shared storage database=] for the responding server's [=/origin=].
+    * `delete`
+        - Required [=structured header/parameters|parameter=]:
+            * `key` ([=structured header/Token=], [=structured header/String=], or [=structured header/Byte Sequence=])
+        - `delete` erases any [=shared storage database/entry=] in the [=shared storage database=] for the responding server's [=/origin=] for which the [=entry/key=] is equal to the [=structured header/parameter=] value of `key`.
+    * `clear`
+        - `clear` erases all [=shared storage database/entries=] in the [=shared storage database=] for the responding server's [=/origin=].
+
+    Note: We allow [=structured header/Byte Sequences=] in order to accommodate Unicode keys and values. Any [=structured header/Byte Sequence=] will be assumed to be [=UTF-8 encoded=] and will fail to parse otherwise.
+
+<span class=todo>Add examples.</span>
+
+## Shared Storage Fetch-Related Algorithms ## {#ss-fetch-algo}
+
+  <div algorithm>
+    To <dfn>determine whether a request can currently use shared storage</dfn>, given a [=/request=] |request|, perform the following steps:
+
+    1. Let |window| to |request|’s [=request/window=].
+    1. If |window| is not an [=environment settings object=] whose [=global object=] is a {{Window}}, return false.
+    1. If the result of running [=determine whether shared storage is allowed=] for |window| is false, return false.
+    1. Let |document| be |window|'s [=global object=]'s [=associated document=].
+    1. Return the result of running [=Is feature enabled in document for origin?=] on "[=PermissionsPolicy/shared-storage=]", |document|, and |request|'s [=request/current URL=]'s [=url/origin=].
```

Comment:
This should probably use [[=Should request be allowed to use feature=]](https://www.w3.org/TR/permissions-policy/#should-request-be-allowed-to-use-feature)? instead of "Is feature enabled...".